ddev/src/ddev/utils/scripts/check_pr.py

```diff
@@ -0,0 +1,189 @@
+"""
+Running this script by itself must not use any external dependencies.
+"""
+# (C) Datadog, Inc. 2023-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from __future__ import annotations
+
+import argparse
+import os
+import re
+import subprocess
+import sys
+from typing import Iterator
+
+
+def requires_changelog(target: str, files: Iterator[str]) -> bool:
+    if target == 'ddev':
+        source = 'src/ddev/'
+    else:
+        if target == 'datadog_checks_base':
+            directory = 'base'
+        elif target == 'datadog_checks_dev':
+            directory = 'dev'
+        elif target == 'datadog_checks_downloader':
+            directory = 'downloader'
+        else:
+            directory = target.replace('-', '_')
+
+        source = f'datadog_checks/{directory}/'
+
+    return any(f.startswith(source) or f == 'pyproject.toml' for f in files)
```

Comment:
Do I understand correctly that this will trigger if I change a docstring in a source file?

Reply:
and also if I perform a refactor that doesn't change behavior. Do we want to enforce changelog entries in this case? I guess one approach to merge despite the validation failing in cases where we manually override this preference/requirement.